cfgov/v1/models/base.py

```diff
@@ -255,6 +267,37 @@ def get_context(self, request, *args, **kwargs):
         if self.schema_json:
             context['schema_json'] = self.schema_json
 
+        context['meta_description'] = ''
+        if self.search_description:
+            context['meta_description'] = self.search_description
+        if hasattr(self, 'header') and not context['meta_description']:
+            context['meta_description'] = self.get_streamfield_content(
+                self.header,
+                'hero',
+                'body')
+        if (
+            hasattr(self, 'preview_description')
```

Comment:
I don't think this needs the `hasattr` constructions, since `preview_description` is a model field, and not part of a StreamField.

Reply:
`preview_description` is a defined model field on the 5 descendants of AbstractFilterPage, but if you try to call it from  another page type (e.g. AnswerPage) it will raise `AttributeError`. If we want this to work globally, the `hasattr` test is needed.

`search_description`, which is defined on wagtail.core.Page, is universal for us, so that doesn't need a test.

Reply:
D'oh, right, sorry.